sklearn/linear_model/_bayes.py

```diff
@@ -179,7 +195,8 @@ class BayesianRidge(RegressorMixin, LinearModel):
     def __init__(
         self,
         *,
-        n_iter=300,
+        n_iter="deprecated",
```

Comment:
Since we are using keyword only argument here, you can move `n_iter` as the last argument and `max_iter` as the first one.